### PR TITLE
feat: implement flareon::main macro for easy framework bootstrapping

### PR DESCRIPTION
**Pull request by m4tx**
This is the first step in making Flareon an actual "framework" rather than just a library. This macro currently doesn't do much now, but eventually we'd like to have an entire CLI that would be automatically generated for each Flareon project and would allow to do some common operations (run server, run migrations, copy static files to a directory, etc.), as well as allow Flareon users to define their own ones.

### CODE REVIEW

File: examples/hello-world/src/main.rs [hunk 1]
```diff
@@ -1,36 +1,9 @@
...
-
-impl FlareonApp for HelloApp {
-    fn name(&self) -> &'static str {
-        env!("CARGO_PKG_NAME")
-    }
-
-    fn router(&self) -> Router {
-        Router::with_urls([Route::with_handler("/", return_hello)])
-    }
-}
```
Comment (seqre) [inline examples/hello-world/src/main.rs:23]: Why was this removed? This hello world no longer hellos the world

File: examples/todo-list/src/main.rs [hunk 2]
```diff
@@ -83,6 +79,16 @@ impl FlareonApp for TodoApp {
         "todo-app"
     }
 
+    fn migrations(&self) -> Vec<Box<dyn DynMigration>> {
+        // TODO: this is way too complicated for the user-facing API
```
Comment (seqre) [inline examples/todo-list/src/main.rs:83]: I agree

File: flareon-macros/tests/ui/attr_main_args.stderr [hunk 3]
```diff
@@ -0,0 +1,11 @@
+error: flareon::main function must have zero arguments
+ --> tests/ui/attr_main_args.rs:2:15
+  |
+2 | async fn main(arg: i32) -> flareon::Result<FlareonProject> {
+  |               ^^^^^^^^
+
+error[E0601]: `main` function not found in crate `$CRATE`
```
Comment (seqre) [inline flareon-macros/tests/ui/attr_main_args.stderr:7]: Is this supposed to be `$CRATE` or is that some mistake?

**Review by seqre — APPROVED** [review]
LGTM after fixing `hello-world` example

File: flareon-macros/tests/ui/attr_main_args.stderr (hunk 3 above)
Comment (m4tx) [reply flareon-macros/tests/ui/attr_main_args.stderr:7]: Yes, the `trybuild` crate [normalizes error messages](https://github.com/dtolnay/trybuild/blob/96340263d9eb0705f98f0fa18b1281d2e9306347/src/normalize.rs#L440).
